### PR TITLE
Submit project form 

### PR DESCRIPTION
**Pull request by jhnguye4**

### CODE REVIEW

File: src/Components/ProjectForm.js [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+    setImageURL(e.target.value);
+  }
+
+  const handleSubmit = event => {
+    const form = new FormData();
+    form.append("productName", name);
+    form.append("productDescription", description);
+    form.append("imageUrl", imageURL);
+    form.append("email", user);
+    Service.post('addProduct', form)
```
Comment (snapcat) [inline src/Components/ProjectForm.js:94]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/CSC510-Group-25/feature-hunt/pullRequest?prid=8358184)
```suggestion
    Service.post("addProduct", form)
```

File: src/Components/ProjectForm.js [hunk 2]
```diff
@@ -0,0 +1,155 @@
+import React from 'react';
+import { ReactSession } from 'react-client-session';
```
Comment (snapcat) [inline src/Components/ProjectForm.js:2]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/CSC510-Group-25/feature-hunt/pullRequest?prid=8358184)
```suggestion
import { ReactSession } from "react-client-session";
```

File: src/Components/Header.js [hunk 3]
```diff
@@ -51,12 +51,14 @@ function Header({setQuery}) {
               />
             </div>
             <div className="links">
+
               <ul data-testid="header_links">
-                <li data-testid="header_sub" onClick={() => window.open('http://tiny.cc/new-project', '_blank')}>Submit Project</li>
+                {loggedin && <li data-testid="header_sub" onClick={goTo('submit-project')}>Submit Project</li>}
```
Comment (snapcat) [inline src/Components/Header.js:56]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/CSC510-Group-25/feature-hunt/pullRequest?prid=8358184)
```suggestion
                {loggedin && <li data-testid="header_sub" onClick={goTo("submit-project")}>Submit Project</li>}
```

File: src/App.js [hunk 4]
```diff
@@ -8,11 +8,13 @@ this file. If not, please write to: featurehuntteam@gmail.com
 import './App.css';
 import {BrowserRouter as Router, Switch, Route} from 'react-router-dom';
 import Product from './Components/Product';
+import ProjectForm from './Components/ProjectForm';
```
Comment (snapcat) [inline src/App.js:11]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/CSC510-Group-25/feature-hunt/pullRequest?prid=8358184)
```suggestion
import ProjectForm from "./Components/ProjectForm";
```

File: src/Components/ProjectForm.js [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+   font-size: 14px;
+   margin: 20px 0px;
+`;
+
+function ProjectForm() {
+  const [name, setName] = React.useState("");
+  const [description, setDescription] = React.useState("");
+  const [imageURL, setImageURL] = React.useState("");
+  const [message, setMessage] = React.useState("");
+  const [user, setUser] = React.useState(['']);
```
Comment (snapcat) [inline src/Components/ProjectForm.js:70]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/CSC510-Group-25/feature-hunt/pullRequest?prid=8358184)
```suggestion
  const [user, setUser] = React.useState([""]);
```

File: src/Components/ProjectForm.js [hunk 6]
```diff
@@ -0,0 +1,155 @@
...
+  }
+
+  const handleSubmit = event => {
+    const form = new FormData();
+    form.append("productName", name);
+    form.append("productDescription", description);
+    form.append("imageUrl", imageURL);
+    form.append("email", user);
+    Service.post('addProduct', form)
+      .then(data =>
```
Comment (snapcat) [inline src/Components/ProjectForm.js:95]: Codacy has a fix for the issue: [Expected parentheses around arrow function argument.](https://app.codacy.com/gh/CSC510-Group-25/feature-hunt/pullRequest?prid=8358184)
```suggestion
      .then((data) =>
```

File: src/Service.js [hunk 7]
```diff
@@ -1,4 +1,5 @@
-const baseUrl = 'http://localhost:5000/';
+ // const baseUrl = 'http://localhost:5000/';
+ const baseUrl = 'https://damp-citadel-25681.herokuapp.com/'
```
Comment (snapcat) [inline src/Service.js:2]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/CSC510-Group-25/feature-hunt/pullRequest?prid=8358184)
```suggestion
 const baseUrl = "https://damp-citadel-25681.herokuapp.com/"
```

File: src/Components/ProjectForm.js [hunk 8]
```diff
@@ -0,0 +1,155 @@
...
+
+  const handleDescriptionChange = (e) => {
+    setDescription(e.target.value);
+  }
+
+  const handleImageURLChange = (e) => {
+    setImageURL(e.target.value);
+  }
+
+  const handleSubmit = event => {
```
Comment (snapcat) [inline src/Components/ProjectForm.js:88]: Codacy has a fix for the issue: [Expected parentheses around arrow function argument.](https://app.codacy.com/gh/CSC510-Group-25/feature-hunt/pullRequest?prid=8358184)
```suggestion
  const handleSubmit = (event) => {
```

File: src/Components/ProjectForm.js [hunk 9]
```diff
@@ -0,0 +1,155 @@
+import React from 'react';
+import { ReactSession } from 'react-client-session';
+import styled from "styled-components";
+import TextField from '@mui/material/TextField';
```
Comment (snapcat) [inline src/Components/ProjectForm.js:4]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/CSC510-Group-25/feature-hunt/pullRequest?prid=8358184)
```suggestion
import TextField from "@mui/material/TextField";
```

File: src/Components/ProjectForm.js [hunk 10]
```diff
@@ -0,0 +1,155 @@
+import React from 'react';
```
Comment (snapcat) [inline src/Components/ProjectForm.js:1]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/CSC510-Group-25/feature-hunt/pullRequest?prid=8358184)
```suggestion
import React from "react";
```

File: src/Service.js (hunk 7 above)
Comment (snapcat) [inline src/Service.js:2]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/CSC510-Group-25/feature-hunt/pullRequest?prid=8358184)
```suggestion
 const baseUrl = "https://damp-citadel-25681.herokuapp.com/"
```

File: src/Components/ProjectForm.js [hunk 11]
```diff
@@ -0,0 +1,162 @@
...
+import { ReactSession } from "react-client-session";
+import styled from "styled-components";
+import TextField from '@mui/material/TextField';
```
Comment (snapcat) [inline src/Components/ProjectForm.js:4]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/CSC510-Group-25/feature-hunt/pullRequest?prid=8358184)
```suggestion
import TextField from "@mui/material/TextField";
```

File: src/Components/ProjectForm.js [hunk 12]
```diff
@@ -0,0 +1,162 @@
+import React from 'react';
```
Comment (snapcat) [inline src/Components/ProjectForm.js:1]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/CSC510-Group-25/feature-hunt/pullRequest?prid=8358184)
```suggestion
import React from "react";
```

File: src/Components/ProjectForm.js [hunk 13]
```diff
@@ -0,0 +1,162 @@
...
+
+  const handleDescriptionChange = (e) => {
+    setDescription(e.target.value);
+  }
+
+  const handleImageURLChange = (e) => {
+    setImageURL(e.target.value);
+  }
+
+  const handleSubmit = event => {
```
Comment (snapcat) [inline src/Components/ProjectForm.js:88]: Codacy has a fix for the issue: [Expected parentheses around arrow function argument.](https://app.codacy.com/gh/CSC510-Group-25/feature-hunt/pullRequest?prid=8358184)
```suggestion
  const handleSubmit = (event) => {
```

File: src/Components/ProjectForm.js (hunk 11 above)
Comment (snapcat) [inline src/Components/ProjectForm.js:4]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/CSC510-Group-25/feature-hunt/pullRequest?prid=8358184)
```suggestion
import TextField from "@mui/material/TextField";
```

File: src/Service.js (hunk 7 above)
Comment (snapcat) [inline src/Service.js:2]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/CSC510-Group-25/feature-hunt/pullRequest?prid=8358184)
```suggestion
 const baseUrl = "https://damp-citadel-25681.herokuapp.com/"
```

File: src/Components/ProjectForm.js (hunk 12 above)
Comment (snapcat) [inline src/Components/ProjectForm.js:1]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/CSC510-Group-25/feature-hunt/pullRequest?prid=8358184)
```suggestion
import React from "react";
```

File: src/Components/ProjectForm.js (hunk 13 above)
Comment (snapcat) [inline src/Components/ProjectForm.js:88]: Codacy has a fix for the issue: [Expected parentheses around arrow function argument.](https://app.codacy.com/gh/CSC510-Group-25/feature-hunt/pullRequest?prid=8358184)
```suggestion
  const handleSubmit = (event) => {
```

File: src/Service.js (hunk 7 above)
Comment (snapcat) [inline src/Service.js:2]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/CSC510-Group-25/feature-hunt/pullRequest?prid=8358184)
```suggestion
 const baseUrl = "https://damp-citadel-25681.herokuapp.com/"
```

File: src/Components/ProjectForm.js (hunk 13 above)
Comment (snapcat) [inline src/Components/ProjectForm.js:88]: Codacy has a fix for the issue: [Expected parentheses around arrow function argument.](https://app.codacy.com/gh/CSC510-Group-25/feature-hunt/pullRequest?prid=8358184)
```suggestion
  const handleSubmit = (event) => {
```

File: src/Components/ProjectForm.js (hunk 12 above)
Comment (snapcat) [inline src/Components/ProjectForm.js:1]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/CSC510-Group-25/feature-hunt/pullRequest?prid=8358184)
```suggestion
import React from "react";
```

File: src/Components/ProjectForm.js (hunk 11 above)
Comment (snapcat) [inline src/Components/ProjectForm.js:4]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/CSC510-Group-25/feature-hunt/pullRequest?prid=8358184)
```suggestion
import TextField from "@mui/material/TextField";
```

File: src/Components/ProjectForm.js (hunk 12 above)
Comment (snapcat) [inline src/Components/ProjectForm.js:1]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/CSC510-Group-25/feature-hunt/pullRequest?prid=8358184)
```suggestion
import React from "react";
```

**Review by peeyush10234 — APPROVED** [review]

File: src/Components/ProjectForm.js (hunk 12 above)
Comment (snapcat) [reply src/Components/ProjectForm.js:1]: Comment by Codacy

**Comment by snapcat** [comment]
Discussion(s):

> jhnguye4 — 10/23/2021
> i have created a project submittal form
> its on group25-submit-project branch
> 
> Leila Moran — 10/23/2021
> Just assigned you to the relevant issue
> 
> jhnguye4 — 10/23/2021
> thanks, but is there multiple issues for this now? because i thought peeyush assigned me it yesterday
> 
> Leila Moran — 10/23/2021
> Uh... could be?
> https://github.com/CSC510-Group-25/feature-hunt/issues/17


> Leila Moran — 11/02/2021
> @ jhnguye4 Do you want to try your hand at improving the pylint scores?
> jhnguye4 — 11/02/2021
> im working right now on trying to get the backend to work with project submittal form right now
> if i have time i will try

> Leila Moran — 11/04/2021
> Added Codacy analysis for good measure, updated README, and did things to the wiki to make MongoDB setup less of a headache.
> 
> Leila Moran — 11/04/2021
> Working on a test suite for ProjectForm.js right now. I'll also fix the broken header tests.
> 
> Leila Moran — 11/04/2021
> Also, there's a potential issue: projects need to have at least one feature so that Feature.js, ProductTile.js, and Product.js don't break and scream about missing indices. To prevent this, a default feature should be added.
> 
> jhnguye4 — 11/04/2021
> peeyush has added that when a project is submitted now
> 
> Leila Moran — 11/04/2021
> It's handled by the backend?
> 
> jhnguye4 — 11/04/2021
> yea its handled where the api is called
> [2:37 AM]
> also tests are failing for the PR could you help with that?
> 
> Leila Moran — 11/04/2021
> Yup, working on it right now
> 
> jhnguye4 — 11/04/2021
> thanks
> 
> Leila Moran — 11/04/2021
> Quick question: are you changing the labels on input, or are you changing the values?
> 
> jhnguye4 — 11/04/2021
> I think we are setting a standard feature when any project is submitted
> 
> Leila Moran — 11/04/2021
> Upvoting is broken on newly added projects. They need to be pushed to the db with a vote count of 1.
> 
> jhnguye4 — 11/04/2021
> Alright, I'll get to that tomorrow then. Thanks for pointing it out
> 
> Leila Moran — 11/04/2021
> Also, the submitted project isn't added to the list of dashboard projects.
> 
> I'm done with the tests. Goodnight!


> Leila Moran — 11/04/2021
> @jhnguye4 I think your branch is ready.
> 
> https://github.com/CSC510-Group-25/feature-hunt/pull/166
> 
> Approve whenever!

**Pull request merged**